static/js/components/dialogs/CollectionFormDialog.js

```diff
@@ -76,28 +78,67 @@ class CollectionFormDialog extends React.Component {
     dispatch(uiActions.setAdminLists(event.target.value));
   };
 
+  validate = () => {
+    const { collectionForm } = this.props;
+    const validation = validateCollectionForm(this.getPayload(collectionForm));
+    this.handleError(validation);
+  };
+
+  getPayload = (collectionForm: CollectionFormState) => {
+    return {
+      title: collectionForm.title,
+      description: collectionForm.description,
+      view_lists: this.calculateListPermissionValue(collectionForm.viewChoice, collectionForm.viewLists),
+      admin_lists: this.calculateListPermissionValue(collectionForm.adminChoice, collectionForm.adminLists),
+    };
+  }
+
   submitForm = async () => {
     const {
       dispatch,
       history,
       collectionUi: { isNew },
-      collectionForm,
+      collectionForm
     } = this.props;
 
-    const payload = {
-      title: collectionForm.title,
-      description: collectionForm.description,
-      view_lists: this.calculateListPermissionValue(collectionForm.viewChoice, collectionForm.viewLists),
-      admin_lists: this.calculateListPermissionValue(collectionForm.adminChoice, collectionForm.adminLists)
-    };
+    const payload = this.getPayload(collectionForm);
 
+    let collection, error;
     if (isNew) {
-      let collection = await dispatch(actions.collectionsList.post(payload));
-      history.push(makeCollectionUrl(collection.key));
+      try {
+        collection = await dispatch(actions.collectionsList.post(payload));
+        history.push(makeCollectionUrl(collection.key));
+      } catch (e) {
+        error = e;
+        this.handleError(e);
+      }
     } else {
-      await dispatch(actions.collections.patch(collectionForm.key, payload));
+      try {
+        await dispatch(actions.collections.patch(collectionForm.key, payload));
+      } catch (e) {
+        error = e;
+        this.handleError(e);
+      }
+    }
+    if (!error) {
+      this.onClose();
     }
+  };
+
+  onClose = ()=> {
+    const { dispatch, hideDialog } = this.props;
     dispatch(uiActions.clearCollectionForm());
+    hideDialog();
+  };
+
+  handleError = async(error: Error) => {
```

Comment:
This doesn't need to be an async function since `dispatch(validateForm(...))` is not asynchronous. Can you remove the `async` here and `await` below?

Reply:
OK